sensor/kubernetes/listener/resources/convert.go

```diff
@@ -58,6 +59,9 @@ type deploymentWrap struct {
 	original         interface{}
 	portConfigs      map[portRef]*storage.PortConfig
 	pods             []*v1.Pod
+	// registryStore is the Sensor's registry store to use.
+	// This is typically only used for testing purposes.
```

Comment:
I think it'd be better if for testing purposes, we just passed the registry options to the calls to populateImageMetadata instead of embedding the store in the deployment wrap

Reply:
I meant this is only non-`nil` for testing. I thought it made more sense to make this an attribute of the deployment wrapper as we are testing a deployment wrapper method but as a parameter to `populateMetadata` makes sense too